omnigibson/simulator.py

```diff
@@ -833,9 +937,26 @@ def step_physics(self):
             Step the physics a single step.
             """
             self._physics_context._step(current_time=self.current_time)
+
+            # Update all APIs
             self._omni_update_step()
             PoseAPI.invalidate()
 
+        def _on_physics_step(self):
+            # Make the controllable object view API refresh
+            ControllableObjectViewAPI.clear()
+            DummyObjectViewAPI.clear()
+
+            # Run the controller step on every controllable object
+            for scene in self.scenes:
+                for obj in scene.objects:
+                    if isinstance(obj, ControllableObject):
```

Comment:
here and elsewhere, would be ideal to have each object class track its own set of objects, so we don't have to naively check through all objects on the stage for every step (also applies to checks for StatefulObjects, etc.)

Reply:
@cgokmen what do you think? I do see this being helpful

Reply:
@cremebrule how hard would it be to add a system into objectregistry that keeps an index of each object by all of its parent classes? e.g. when you add a DatasetObject X into the registry it will get added into lists for DatasetObject, UsdObject, ... EntityPrim, ... etc. that way the tracking happens on the registry and is addressed upon addition and removal. I am opposed to global state tracking on classes.

Reply:
Either way we should do this in a follow-up imo. Good starter issue for Frank. @hang-yin please open a ticket